scripts/crypto.js

```diff
@@ -353,120 +339,74 @@ function verifyDleq(bp, chkM, chkZ) {
     return true;
 }
 
-// Check that the underlying DLEQ proof is well-defined
-function isDleqCompleteAndSane(dleq, chkM, chkZ, proofC) {
-    if (!dleq.M || !dleq.Z || !dleq.R || !dleq.C) {
-        console.error(DLEQ_PROOF_INCOMPLETE);
-        return false;
-    }
-
-    // Check that all points are on the same curve
-    let curveG = dleq.G.curve;
-    let curveH = dleq.H.curve;
-    let curveM = dleq.M.curve;
-    let curveZ = dleq.Z.curve;
-    if (sjcl.ecc.curveName(curveG) != sjcl.ecc.curveName(curveH) ||
-        sjcl.ecc.curveName(curveH) != sjcl.ecc.curveName(curveM) ||
-        sjcl.ecc.curveName(curveM) != sjcl.ecc.curveName(curveZ) ||
-        sjcl.ecc.curveName(curveG) != P256_NAME) {
-        console.error(INCORRECT_CURVE_ERR);
-        return false;
-    }
-
-    let chkMPoint;
-    let chkZPoint;
+// Recompute the composite M and Z values for verifying DLEQ
+function recomputeComposites(chkM, chkZ) {
+    let seed = getSeedPRNG(chkM, chkZ);
+    let shake = createShake256();
+    shake.update(seed, "hex");
+    let cM;
+    let cZ;
     for (let i=0; i<chkM.length; i++) {
-        let cMi = _scalarMult(proofC[i], chkM[i].point);
-        let cZi = _scalarMult(proofC[i], chkZ[i]);
-
-        if (!chkMPoint && !chkZPoint) {
-            chkMPoint = cMi;
-            chkZPoint = cZi;
+        let ci = getShakeScalar(shake);
+        let cMi = _scalarMult(ci, chkM[i].point);
+        let cZi = _scalarMult(ci, chkZ[i]);
+        if (cM === undefined || cZ === undefined) {
+            cM = cMi;
+            cZ = cZi;
         } else {
-            chkMPoint = chkMPoint.toJac().add(cMi).toAffine();
-            chkZPoint = chkZPoint.toJac().add(cZi).toAffine();
+            cM = cM.toJac().add(cMi).toAffine();
+            cZ = cZ.toJac().add(cZi).toAffine();
         }
     }
-    if (!sjcl.bitArray.equal(dleq.M.toBits(), chkMPoint.toBits()) || !sjcl.bitArray.equal(dleq.Z.toBits(), chkZPoint.toBits())) {
-        console.error(INCONSISTENT_DLEQ_PROOF_ERR);
-        return false;
-    }
-    return true;
+
+    return {M: cM, Z: cZ};
 }
 
-// Checks that the batch proof is well-defined
-function isBatchProofCompleteAndSane(bp, chkM, chkZ) {
```

Comment:
Nice. This gist of this deletion is that we don't need to check the point formation since we just verify the proof, right?

Reply:
Yes that's right. We used to send all the proof values and check they were the same after recomputation. Then we verified the proof using the values that were sent by the server. This is a suboptimal approach and not as the protocol is described, exactly.

We now just verify the proof directly using only the (c,s) values as detailed in the protocol.